File: WalletTracker/app/src/main/java/com/chtima/wallettracker/dao/TransactionDao.java
package com.chtima.wallettracker.dao;

import androidx.room.*;

import com.chtima.wallettracker.models.Transaction;

import java.util.List;

import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.core.Maybe;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface TransactionDao {
    @Insert
    Single<Long> insert (Transaction transaction);

    @Query("SELECT * FROM transactions")
    Flowable<List<Transaction>> getAll();
}

File: WalletTracker/app/src/main/java/com/chtima/wallettracker/db/dao/TransactionDao.kt
package com.chtima.wallettracker.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.chtima.wallettracker.models.Transaction
import io.reactivex.rxjava3.core.Flowable
import io.reactivex.rxjava3.core.Single

@Dao
interface TransactionDao {
    @Insert
    fun insert(transaction: Transaction): Single<Long>

    @Query("SELECT * FROM `transaction`")
    fun getAll(): Flowable<List<Transaction>>
}